commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2024 Marco Collovati, Dario Götze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mcollovati.quarkus.hilla.deployment;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Scanner;
+
+import com.vaadin.hilla.Endpoint;
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.deployment.CodeGenContext;
+import io.quarkus.deployment.CodeGenProvider;
+import org.eclipse.microprofile.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TypescriptClientCodeGenProvider implements CodeGenProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TypescriptClientCodeGenProvider.class);
+
+    static final String FRONTEND_FOLDER_NAME = "frontend";
+    private Path frontendFolder;
+
+    @Override
+    public String providerId() {
+        return "quarkus-hilla-connect-client";
+    }
+
+    @Override
+    public String inputDirectory() {
+        return FRONTEND_FOLDER_NAME;
+    }
+
+    @Override
+    public void init(ApplicationModel model, Map<String, String> properties) {
+        if (model.getApplicationModule() == null) {
+            LOGGER.info("ApplicationModule is null");
+            return;
+        }
+        File moduleDir = model.getApplicationModule().getModuleDir();
+        Path legacyFrontendFolder = moduleDir.toPath().resolve(FRONTEND_FOLDER_NAME);
+        Path frontendDir = moduleDir.toPath().resolve(Path.of("src", "main", FRONTEND_FOLDER_NAME));
+        if (Files.isDirectory(frontendDir)) {
+            this.frontendFolder = frontendDir;
+            LOGGER.debug("Using frontend folder {}", frontendDir);
+        } else if (Files.isDirectory(legacyFrontendFolder)) {
+            this.frontendFolder = legacyFrontendFolder;
+            LOGGER.debug("Using legacy frontend folder {}", legacyFrontendFolder);
+        } else {
+            LOGGER.debug("Frontend folder not detected");
```

Comment:
```suggestion
            LOGGER.debug("Frontend folder not found");
```
Just my preferred naming.